### ISSUE TITLE
Failover node isn't detected on Active/Passive storage

### ISSUE THREAD

**Issue by AlexTalker** [whamcloud/iml-agent#67]
Hi @jgrund @utopiabound 

I meet a certain problem with filesystem detection.
I have a double-controller storage system where block devices either in active or passive state.
Filesystem created on a block device that is available only if its state is 'active'.
Trouble it that your `Primary/Failover` node detection is based on a fact that filesystem metadata can at least be read from block devices which isn't my case. However, fail-over information specified in filesystem configuration do contain address of the 'passive' node.
My question that is there is a way to detect filesystems in such configurations with successful determination of a fail-over node?
I don't know if this question is more related to UI or the agent, so feel free to point out it for me.

**Comment by utopiabound** [comment]
Are you using multipathd?
Is your storage cross-connected to both hosts? or is one host connected to controller A and the other to controller B?

**Comment by utopiabound** [comment]
Do the block devices show up with the same SCSI id?  If it's really just a replica and has a separate SCSI id, I don't see how we could discover it.

**Comment by AlexTalker** [comment]
It is not a SCSI device I think, so I doubt you can get SCSI id for it but block device on 'Active' node discovered 'okay'. Problem is just that on passive you can't read any data from it. But the same path in `/dev` exists as well. And as I said, the `Fail-over` node address specified in filesystem settings.
Also, `Monitoring` mode of this product used on both nodes, not `Management`.

**Comment by AlexTalker** [comment]
@jgrund A volume is only showing a primary host in the GUI.
As far as I get it, by the reason I described above.

**Comment by utopiabound** [comment]
How would we detect it's the fail-over volume?

**Comment by AlexTalker** [comment]
@jgrund A volume is only showing a primary host in the GUI.
As far as I get it, by the reason I described above.

**Comment by AlexTalker** [comment]
@utopiabound you can check the filesystem properties(about fail-over node) on active block device and compare with metadata regarding `LNet`

**Comment by jgrund** [comment]
@AlexTalker I'm not totally familiar with the types of devices you are using, but if they are in /dev, perhaps they have some useful udev info as well.

Could you run:

```sh
 echo '"Stream"' | socat - UNIX-CONNECT:/var/run/device-scanner.sock | jq
```

on an active and passive node that have IML agents on them and post output? You might need to install `socat` and `jq` first.

If it's easier you could send output to iml@whamcloud.com too.

**Comment by AlexTalker** [comment]
I did created related [`PR`](https://github.com/whamcloud/integrated-manager-for-lustre/pull/788)
I think now that problem is on the GUI side so I'll continue hacking it(I already succeeded with not-backed-by-zfs volumes but the solution is so dirty)
So I think this issue should actually be related to GUI, since once I made a workaround to passive device to be read-only, the agent detect everything well.

**Comment by AlexTalker** [comment]
@jgrund Am I correct to assume that you can't call `zfs get` on `fail-over node` if backed by `ZFS` volume is imported on `primary note`?
I found out that if I do allow data to be read from passive node block devices, the only milestone on my way is the GUI-related issue with `NIDs` and backed by `ZFS` volumes.
How do you detect `fail-over` node for the `ZFS`-backed volumes if `Lustre` on them can be detected on one node at a time?

**Comment by jgrund** [comment]
@AlexTalker correct, we don't want to do double imports of zpools to discover device state.

> How do you detect fail-over node for the ZFS-backed volumes if Lustre on them can be detected on one node at a time?

This is done by id'ing block devices on separate nodes as the same via SCSI ID.

In the case of "composite" devices such as zpools we check the underlying vdev ids to determine if a pool can be shared between hosts.
